### PR TITLE
feature(changelog): Make changelog create from --type flag (canary or release)

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

File: build-system/tasks/changelog.js [hunk 1]
```diff
@@ -117,25 +128,36 @@ function buildChangelog(gitMetadata, githubMetadata) {
...
 }
 
+/**
+ * @param {string} tag
+ * @return {!Promise<string>}
+ */
 function getGitLog(tag) {
   var options = {
-    args: 'log release...' + tag + ' --pretty=format:%s --merges'
+    args: 'log ' + branch + '...' + tag + ' --pretty=format:%s --merges'
```
Comment (jridgewell) [inline build-system/tasks/changelog.js]: Template strings would help here and below.

Comment (erwinmombay) [inline build-system/tasks/changelog.js]: for sure, you happen to know is it supported on node 4.0? haven't migrated our build-system files to es6 yet. (we dont even use arrow functions there yet)

Comment (jridgewell) [inline build-system/tasks/changelog.js]: Yup, they're in there.

Comment (erwinmombay) [inline build-system/tasks/changelog.js]: @jridgewell thanks. will make the updates as i update this file (will be making some more this week)

**Pull request merged**